Road/source/features/feature_selected.py

```diff
@@ -0,0 +1,66 @@
+# Standard library imports
+import ast
+import sys
+import os
+
+# Related third party imports
+import pandas as pd
+
+# Reset the path to the parent folder to import any mudule we want
+target = os.path.abspath(__file__)
+while (target.split("\\")[-1] != "Road"):
+    target = os.path.dirname(target)
+sys.path.append(target)
+
+# Local imports
+from source.constants import COLUMNS_TO_KEEP
+
+
+def delete_unnecessery_columns(data: pd.DataFrame) -> None:
+    """Remove all unnecessary columns except id, date, message,
+    from_id and reply_to columns.
+
+    Args:
+        data (pd.DataFrame): The dataframe that will be modify.
+
+    Returns:
+        None
+    """
+    return data[COLUMNS_TO_KEEP].copy()
+
+def extract_dates(date: pd.Series) -> None:
+    """Extract year, month, day, hour, minute, and second values
+    from the date series.
+
+    Args:
+        date (pd.Series): The date column to turn it into useful columns
+
+    Returns:
+        None
+    """
+    date = pd.to_datetime(date)
+    year =date.dt.year
+    month =date.dt.month
+    day =date.dt.day
+    hour =date.dt.hour
+    minute =date.dt.minute
+    second =date.dt.second
+
+    return (year, month, day, hour, minute, second)
+
+
+def extract_id(json_string: str, target_id: str) -> int:
```

Comment:
Where will we use this function?

Reply:
This file is not related to this PR, its related to Data Preparation PR.
The 'extract id' function is used to extract the ID from two required JSON-formatted columns which are `from_id` and `reply_to` columns.